src/Microsoft.TemplateEngine.Cli/Commands/InstantiateCommand.TabCompletion.cs

```diff
@@ -11,27 +12,36 @@ namespace Microsoft.TemplateEngine.Cli.Commands
 {
     internal partial class InstantiateCommand : BaseCommand<InstantiateCommandArgs>
     {
-        internal IEnumerable<string> GetSuggestions(
+        internal IEnumerable<CompletionItem> GetCompletions(
             InstantiateCommandArgs args,
             IEnumerable<TemplateGroup> templateGroups,
             IEngineEnvironmentSettings environmentSettings,
             TemplatePackageManager templatePackageManager,
-            string? textToMatch)
+            TextCompletionContext context)
         {
             if (string.IsNullOrWhiteSpace(args.ShortName))
             {
-                return templateGroups.SelectMany(g => g.ShortNames).Distinct().OrderBy(s => s, StringComparer.OrdinalIgnoreCase);
+                return templateGroups
+                    .SelectMany(g => g.ShortNames, (g, shortName) => new CompletionItem(shortName, documentation: g.Description))
+                    .Distinct()
+                    .OrderBy(c => c.Label, StringComparer.OrdinalIgnoreCase)
+                    .Concat(base.GetCompletions(context, environmentSettings));
             }
             else
             {
                 //no exact match on short name
                 if (!templateGroups.Any(template => template.ShortNames.Contains(args.ShortName)))
                 {
-                    return templateGroups.SelectMany(g => g.ShortNames).Where(n => n.StartsWith(args.ShortName)).Distinct().OrderBy(s => s, StringComparer.OrdinalIgnoreCase);
+                    return templateGroups
+                        .SelectMany(g => g.ShortNames, (g, shortName) => new CompletionItem(shortName, documentation: g.Description))
+                        .Where(c => c.Label.StartsWith(args.ShortName))
+                        .Distinct()
+                        .OrderBy(c => c.Label, StringComparer.OrdinalIgnoreCase)
+                        .Concat(base.GetCompletions(context, environmentSettings));
```

Comment:
Can `base.Getcompletions` provide already existing completion items? If yes, we might want to call `Distinct` after `Concat` to eliminate all the duplicates.

Reply:
no, it cannot. `Concat` will provide suggestions for subcommands / help option, whereas 1st part is providing suggestions for template short name. 

There is a very specific case when template short name is same as subcommand, but I don't think we care for it for completion.